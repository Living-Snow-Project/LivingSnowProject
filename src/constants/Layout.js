import { Dimensions } from "react-native";

const { height, width } = Dimensions.get("window");

export default {
  window: {
    width,
    height,
  },
  isSmallDevice: width < 375,
};
