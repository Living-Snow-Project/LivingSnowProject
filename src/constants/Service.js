export const serviceEndpoint = 'https://snowalgaeproductionapp.azurewebsites.net';
