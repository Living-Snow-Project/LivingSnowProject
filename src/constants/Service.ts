const serviceEndpoint = "https://snowalgaeproductionapp.azurewebsites.net";
const photosBlobStorageEndpoint =
  "https://snowalgaestorage.blob.core.windows.net/photos";

export { serviceEndpoint, photosBlobStorageEndpoint };
