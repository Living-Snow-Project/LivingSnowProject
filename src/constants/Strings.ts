// for TextInput placeholder prop
const Placeholders = {
  GPS: {
    AcquiringLocation: "Looking for GPS signal",
    EnterCoordinates: "ie. 12.345678, -123.456789",
    NoPermissions:
      "Check location permissions in Settings. Enter coordinates manually.",
    NoLocation: "Could not determine location. Enter coordinates manually.",
  },
  // TODO: RecordScreen placeholders
};

const Labels = {
  TimelineScreen: {
    DownloadedRecords: "Downloaded",
    NoRecords: "No records to display",
    PendingRecords: "Pending",
  },
  StatusBar: {
    NoConnection: "No Internet Connection",
  },
};

// for Alert.alert calls
const Notifications = {
  uploadSuccess: {
    title: "Upload succeeded",
    message: "Thanks for your submission.",
  },
  uploadFailed: {
    title: "Record Saved",
    message: "We will upload it later.",
  },
  invalidCoordinates: {
    title: "Invalid GPS coordinates",
    message:
      'Coordinates must be in "lat, long" format. ie. 12.345678, -123.456789',
  },
};

export { Notifications, Placeholders, Labels };
