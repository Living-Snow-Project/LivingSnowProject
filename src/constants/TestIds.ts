const TestIds = {
  GPS: {
    gpsManualInputTestId: "gps-manual-input",
    gpsManualPressableTestId: "gps-manual-pressable",
  },
  Photos: {
    photoSelectorTestId: "photo-selector",
  },
  Pickers: {
    recordSelectorTestId: "record-type-picker",
    atlastSelectorTestId: "atlas-type-picker",
  },
  RecordScreen: {
    UploadButton: "record-screen-upload-button",
  },
  TimelineScreen: {
    RefreshControl: "timeline-screen-refresh-control",
    SettingsButton: "timeline-screen-settings-button",
    NewRecordButton: "timeline-screen-record-button",
  },
  SettingsScreen: {
    ShowGpsWarning: "settings-screen-show-gps-warning",
    ShowAtlasRecords: "settings-screen-show-atlas-records",
    ShowOnlyAtlasRecords: "settings-screen-show-only-atlas-records",
  },
  Icons: {
    SampleIcon: "sample-icon",
    SightingIcon: "sighting-icon",
    PictureIcon: "picture-icon",
    DeleteIcon: "delete-icon",
    EditIcon: "edit-icon",
    SnowIcon: "snow-icon",
  },
  RecordList: {
    EditRecordAction: "edit-record-action",
    DeleteRecordAction: "delete-record-action",
  },
};

export default TestIds;
