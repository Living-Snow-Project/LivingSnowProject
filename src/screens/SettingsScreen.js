import React from "react";
import { StyleSheet, Switch, Text, TextInput, View } from "react-native";
import { Storage } from "../lib/Storage";

export default class SettingsScreen extends React.Component {
  static navigationOptions = {
    title: "Settings",
    headerTitleContainerStyle: { justifyContent: "center" },
  };

  toggleShowGpsWarning(value) {
    global.appConfig.showGpsWarning = value;
    Storage.saveAppConfig();
    this.forceUpdate();
  }

  toggleShowAtlasRecords(value) {
    global.appConfig.showAtlasRecords = value;
    Storage.saveAppConfig();
    this.forceUpdate();
  }

  toggleShowOnlyAtlasRecords(value) {
    global.appConfig.showOnlyAtlasRecords = value;
    Storage.saveAppConfig();
    this.forceUpdate();
  }

  updateName(name) {
    global.appConfig.name = name;
    Storage.saveAppConfig();
    this.forceUpdate();
  }

  updateOrganization(organization) {
    global.appConfig.organization = organization;
    Storage.saveAppConfig();
    this.forceUpdate();
  }

  render() {
    return (
      <View>
        <Text style={styles.optionStaticText}>Name</Text>
        <TextInput
          style={styles.optionInputText}
          value={global.appConfig.name}
          onChangeText={(name) => this.updateName(name)}
          maxLength={50}
          returnKeyType="done"
        />

        <Text style={styles.optionStaticText}>Organization</Text>
        <TextInput
          style={styles.optionInputText}
          value={global.appConfig.organization}
          onChangeText={(organization) => this.updateOrganization(organization)}
          maxLength={50}
          returnKeyType="done"
        />

        <Text style={styles.optionStaticText}>Notifications</Text>
        <View style={styles.optionContainer}>
          <Text style={styles.optionStaticText}>
            Show Manual Coordinates Warning
          </Text>
          <Switch
            style={styles.switch}
            onValueChange={(value) => this.toggleShowGpsWarning(value)}
            value={global.appConfig.showGpsWarning}
          />
        </View>

        <Text style={styles.optionStaticText}>Snow Algae Atlas</Text>
        <View style={styles.optionContainer}>
          <Text style={styles.optionStaticText}>Show Atlas Records</Text>
          <Switch
            style={styles.switch}
            disabled={global.appConfig.showOnlyAtlasRecords}
            onValueChange={(value) => this.toggleShowAtlasRecords(value)}
            value={global.appConfig.showAtlasRecords}
          />
        </View>
        <View style={[styles.optionContainer, { marginTop: 5 }]}>
          <Text style={styles.optionStaticText}>Show Only Atlas Records</Text>
          <Switch
            style={styles.switch}
            disabled={!global.appConfig.showAtlasRecords}
            onValueChange={(value) => this.toggleShowOnlyAtlasRecords(value)}
            value={global.appConfig.showOnlyAtlasRecords}
          />
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  optionContainer: {
    backgroundColor: "#efefef",
    borderWidth: 1,
    borderRadius: 4,
    borderColor: "gray",
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    marginLeft: 10,
    marginRight: 10,
    paddingVertical: 10,
  },
  optionTextContainer: {
    paddingHorizontal: 4,
    flexDirection: "row",
    alignItems: "center",
  },
  optionStaticText: {
    fontSize: 15,
    paddingLeft: 10,
    paddingVertical: 2,
  },
  optionInputText: {
    backgroundColor: "#efefef",
    paddingHorizontal: 15,
    paddingVertical: 15,
    borderWidth: 1,
    borderColor: "gray",
    borderRadius: 4,
    marginLeft: 10,
    marginRight: 10,
  },
  switch: {
    marginRight: 10,
    borderColor: "black",
  },
});
