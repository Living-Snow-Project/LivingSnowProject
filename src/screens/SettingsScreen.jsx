import React, { useContext } from "react";
import { Switch, Text, View } from "react-native";
import styles from "../styles/Settings";
import UserIdentityInput from "../components/forms/UserIdentityInput";
import { AppSettingsContext } from "../../AppSettings";

export default function SettingsScreen() {
  const {
    showGpsWarning,
    showAtlasRecords,
    showOnlyAtlasRecords,
    updateAppSettings,
  } = useContext(AppSettingsContext);
  return (
    <View>
      <UserIdentityInput />

      <Text style={styles.optionStaticText}>Notifications</Text>
      <View style={styles.optionContainer}>
        <Text style={styles.optionStaticText}>
          Show Manual Coordinates Warning
        </Text>
        <Switch
          style={styles.switch}
          onValueChange={(value) =>
            updateAppSettings({ showGpsWarning: value })
          }
          value={showGpsWarning}
        />
      </View>

      <Text style={styles.optionStaticText}>Snow Algae Atlas</Text>
      <View style={styles.optionContainer}>
        <Text style={styles.optionStaticText}>Show Atlas Records</Text>
        <Switch
          style={styles.switch}
          disabled={showOnlyAtlasRecords}
          onValueChange={(value) =>
            updateAppSettings({ showAtlasRecords: value })
          }
          value={showAtlasRecords}
        />
      </View>
      <View style={[styles.optionContainer, { marginTop: 5 }]}>
        <Text style={styles.optionStaticText}>Show Only Atlas Records</Text>
        <Switch
          style={styles.switch}
          disabled={!showAtlasRecords}
          onValueChange={(value) =>
            updateAppSettings({ showOnlyAtlasRecords: value })
          }
          value={showOnlyAtlasRecords}
        />
      </View>
    </View>
  );
}
