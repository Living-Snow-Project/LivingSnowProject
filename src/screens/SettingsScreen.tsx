import React, { useState } from "react";
import { Switch, Text, View } from "react-native";
import styles from "../styles/Settings";
import UserIdentityInput from "../components/forms/UserIdentityInput";
import { getAppSettings, setAppSettings } from "../../AppSettings";
import TestIds from "../constants/TestIds";

export default function SettingsScreen() {
  const [{ showGpsWarning }, setSettings] = useState(getAppSettings());

  return (
    <>
      <UserIdentityInput />

      <Text style={styles.optionStaticText}>Notifications</Text>
      <View style={styles.optionContainer}>
        <Text style={styles.optionStaticText}>
          Show Manual Coordinates Warning
        </Text>
        <Switch
          testID={TestIds.SettingsScreen.ShowGpsWarning}
          style={styles.switch}
          onValueChange={(value) => {
            setSettings((prev) => {
              setAppSettings({ ...prev, showGpsWarning: value });
              return { ...prev, showGpsWarning: value };
            });
          }}
          value={showGpsWarning}
        />
      </View>
    </>
  );
}
