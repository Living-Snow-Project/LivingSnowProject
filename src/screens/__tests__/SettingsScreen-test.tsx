import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import SettingsScreen from "../SettingsScreen";
import { Placeholders } from "../../constants/Strings";
import { getAppSettings } from "../../../AppSettings";

describe("SettingsScreen test suite", () => {
  test("renders", () => {
    const { toJSON } = render(<SettingsScreen />);

    expect(toJSON()).toMatchSnapshot();
  });

  test("user name", async () => {
    const { getByDisplayValue, getByPlaceholderText } = render(
      <SettingsScreen />
    );
    const username = getByPlaceholderText(Placeholders.Settings.Username);
    const expected = "Test User Name";

    fireEvent.changeText(username, expected);
    expect(getByDisplayValue(expected)).toBeTruthy();
    expect(getAppSettings().name).toEqual(expected);
    fireEvent(username, "onSubmitEditing");
  });

  test("user organization", () => {
    const { getByDisplayValue, getByPlaceholderText } = render(
      <SettingsScreen />
    );
    const organization = getByPlaceholderText(
      Placeholders.Settings.Organization
    );
    const expected = "Test Organization";

    fireEvent.changeText(organization, expected);
    expect(getByDisplayValue(expected)).toBeTruthy();
    expect(getAppSettings().organization).toEqual(expected);
    fireEvent(organization, "onSubmitEditing");
  });

  test.todo("toggle showGpsWarning");
  test.todo("toggle showAtlasRecords");
  test.todo("toggle showOnlyAtlastRecords");
});
