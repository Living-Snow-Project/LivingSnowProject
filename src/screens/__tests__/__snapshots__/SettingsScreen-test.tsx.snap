// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsScreen test suite renders 1`] = `
Array [
  <Text
    style={
      Object {
        "fontSize": 15,
        "paddingLeft": 10,
        "paddingVertical": 2,
      }
    }
  >
    Name
  </Text>,
  <TextInput
    allowFontScaling={true}
    maxLength={50}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Enter your name"
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#efefef",
        "borderColor": "gray",
        "borderRadius": 4,
        "borderWidth": 1,
        "marginLeft": 10,
        "marginRight": 10,
        "paddingHorizontal": 15,
        "paddingVertical": 15,
      }
    }
    underlineColorAndroid="transparent"
  />,
  <Text
    style={
      Object {
        "fontSize": 15,
        "paddingLeft": 10,
        "paddingVertical": 2,
      }
    }
  >
    Organization
  </Text>,
  <TextInput
    allowFontScaling={true}
    maxLength={50}
    onChangeText={[Function]}
    placeholder="Enter the organization you belong to (if any)"
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#efefef",
        "borderColor": "gray",
        "borderRadius": 4,
        "borderWidth": 1,
        "marginLeft": 10,
        "marginRight": 10,
        "paddingHorizontal": 15,
        "paddingVertical": 15,
      }
    }
    underlineColorAndroid="transparent"
  />,
  <Text
    style={
      Object {
        "fontSize": 15,
        "paddingLeft": 10,
        "paddingVertical": 2,
      }
    }
  >
    Notifications
  </Text>,
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#efefef",
        "borderColor": "gray",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginLeft": 10,
        "marginRight": 10,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 15,
          "paddingLeft": 10,
          "paddingVertical": 2,
        }
      }
    >
      Show Manual Coordinates Warning
    </Text>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "height": 31,
            "width": 51,
          },
          Object {
            "borderColor": "black",
            "marginRight": 10,
          },
        ]
      }
      value={true}
    />
  </View>,
  <Text
    style={
      Object {
        "fontSize": 15,
        "paddingLeft": 10,
        "paddingVertical": 2,
      }
    }
  >
    Snow Algae Atlas
  </Text>,
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#efefef",
        "borderColor": "gray",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginLeft": 10,
        "marginRight": 10,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 15,
          "paddingLeft": 10,
          "paddingVertical": 2,
        }
      }
    >
      Show Atlas Records
    </Text>
    <RCTSwitch
      accessibilityRole="switch"
      disabled={false}
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "height": 31,
            "width": 51,
          },
          Object {
            "borderColor": "black",
            "marginRight": 10,
          },
        ]
      }
      value={false}
    />
  </View>,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#efefef",
          "borderColor": "gray",
          "borderRadius": 4,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginLeft": 10,
          "marginRight": 10,
          "paddingVertical": 10,
        },
        Object {
          "marginTop": 5,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "fontSize": 15,
          "paddingLeft": 10,
          "paddingVertical": 2,
        }
      }
    >
      Show Only Atlas Records
    </Text>
    <RCTSwitch
      accessibilityRole="switch"
      disabled={true}
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "height": 31,
            "width": 51,
          },
          Object {
            "borderColor": "black",
            "marginRight": 10,
          },
        ]
      }
      value={false}
    />
  </View>,
]
`;
